### PR TITLE
add the box item

### PR DESCRIPTION
**Pull request by pochretien**
add uts for the box item
add examples for the box item
dont show the tooltip if the title is empty
add uts for the modif on the tooltip title

### CODE REVIEW

**Comment by pochretien** [comment]
image
image

File: src/components/boxItem/BoxItem.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        'hidden': this.props.hidden,
+      },
+      this.props.classes);
+  }
+
+  protected getDropdownPrepend(prefix: string): JSX.Element {
+    if (prefix) {
+      return <span className='text-medium-grey mr1'>{prefix}</span>;
+    }
+    return null;
```
Comment (sroy3) [inline src/components/boxItem/BoxItem.tsx]: `return prefix ? <span className='text-medium-grey mr1'>{prefix}</span> : null;`

File: src/components/boxItem/BoxItem.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  protected getSvg(svg: ISvgProps): JSX.Element {
+    if (svg) {
+      return (<Svg {..._.defaults(svg, { className: 'inline-flex mr1', svgClass: 'icon' }) } />);
+    }
+    return null;
```
Comment (sroy3) [inline src/components/boxItem/BoxItem.tsx]: `return svg ? <Svg {..._.defaults(svg, { className: 'inline-flex mr1', svgClass: 'icon' }) } /> : null;`

File: src/components/boxItem/BoxItem.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  handleOnOptionClick = (e: React.MouseEvent<HTMLSpanElement>) => {
+    if (e.target) {
+      if (this.props.onOptionClick) {
```
Comment (sroy3) [inline src/components/boxItem/BoxItem.tsx]: Merge the 2 if statements

File: src/components/boxItem/BoxItem.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  protected getSvg(svg: ISvgProps): JSX.Element {
+    if (svg) {
+      return (<Svg {..._.defaults(svg, { className: 'inline-flex mr1', svgClass: 'icon' }) } />);
+    }
+    return null;
+  }
+
+  handleOnOptionClick = (e: React.MouseEvent<HTMLSpanElement>) => {
+    if (e.target) {
```
Comment (sroy3) [inline src/components/boxItem/BoxItem.tsx]: When is `e.target` not defined?

File: src/components/boxItem/tests/BoxItem.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        { attachTo: document.getElementById('App') },
+      );
+    };
+
+    it('should render the display value', () => {
+      const displayValue: string = 'display value';
+      renderBoxItem({
+        displayValue,
+      });
+      expect(boxItemComponent.find('li').text()).toEqual(displayValue);
```
Comment (sroy3) [inline src/components/boxItem/tests/BoxItem.spec.tsx]: `toBe` instead of `toEqual`

File: src/components/boxItem/BoxItem.tsx (hunk 3 above)
Comment (pochretien) [reply src/components/boxItem/BoxItem.tsx]: validation removed, I dont need the e, I use the this.props :)

File: src/components/itemBox/ItemBox.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+export class ItemBox extends React.Component<IItemBoxProps, void> {
+
+  static defaultProps: Partial<IItemBoxProps> = {
+    tooltip: {
+      title: '',
+    },
+  };
+
+  getClasses(): string {
```
Comment (GermainBergeron) [inline src/components/itemBox/ItemBox.tsx]: private?

File: src/components/itemBox/ItemBox.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      this.props.onOptionClick(this.props);
+    }
+  }
+
+  render() {
+    return (
+      <Tooltip {...this.props.tooltip}>
+        <li
+          className={this.getClasses()}
+          onMouseDown={() => this.handleOnOptionClick()}
```
Comment (GermainBergeron) [inline src/components/itemBox/ItemBox.tsx]: onClick instead of mouse down?

File: src/components/itemBox/examples/ItemBoxExamples.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      placement: 'bottom',
+      container: 'body',
+    };
+
+    const svg: ISvgProps = {
+      svgName: 'domain-google',
+      svgClass: 'icon',
+    };
+    const getSvgPrepend: IContentProps = { content: () => <Svg {...svg} />, classes: ['mr1'] };
+    const getSvgAppend: IContentProps = { content: () => <Svg {...svg} />, classes: ['ml1'] };
```
Comment (GermainBergeron) [inline src/components/itemBox/examples/ItemBoxExamples.tsx]: rename for `svgPrepend` and `svgAppend` as it's not a getter but a simple var

File: src/components/tooltip/Tooltip.tsx [hunk 9]
```diff
@@ -81,12 +83,15 @@ export class Tooltip extends React.Component<ITooltipProps, any> {
...
+    if (this.props.title) {
+      return (
+        <OverlayTrigger {..._.omit(this.props, OVERLAY_PROPS_TO_OMIT) } overlay={tooltip}>
+          <span className={this.props.className}>
+            {this.props.children}
+          </span>
+        </OverlayTrigger>
+      );
+    }
+    return (<span>{this.props.children}</span>);
```
Comment (GermainBergeron) [inline src/components/tooltip/Tooltip.tsx]: Still need the span? If you do, should you add the same className as the span in the overlay?

File: src/components/itemBox/ItemBox.tsx (hunk 7 above)
Comment (pochretien) [reply src/components/itemBox/ItemBox.tsx]: I got a problem with the onClick when we try to use it with other component like dropdown. Its a problem with the event order with the onBlur

File: src/components/tooltip/Tooltip.tsx (hunk 9 above)
Comment (pochretien) [reply src/components/tooltip/Tooltip.tsx]: span removed

File: src/components/content/Content.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  static defaultProps: Partial<IContentProps> = {
+    classes: [],
+  };
+
+  private getContent(): JSX.Element | string {
+    return _.isString(this.props.content) ? this.props.content : <this.props.content />;
+  }
+
+  render() {
+    return (<span className={this.props.classes.join(' ')}>
```
[the remark this answers is not in the document]
Comment (GermainBergeron) [reply src/components/content/Content.tsx]: What don't you like about the package? I would use it everywhere in react-vapor to have only one way of doing things. The lib is quite nice when you need to set some class depending on some booleans or having a default class and still having the classes props.

If we don't use it we need to set the defaultProps and use a join, things that we then need to add tests for. The library is already tested which is great

**Pull request merged**